Update EEG Utils & NBS_BCT Utils

maddoxx02 opened the pull request:
- Added EEG Files (test_eeg_utils, eeg_utils, & eeg data .csv) 
- Updated nbs_utils.py (performs explicit thresholding for comparision with TFNOS) 
- Updated example notebook with minor figure changes. 


Pending: 
- Add comments 
- Build ReadMe + graphical abstract
- Add link to Datasets
- Revision of all docStrings 
- Improve example.notebook + add markdowns with comments
- Custom Build Docs
- Pip package 
- Push Final

maddoxx02 committed:
update

maddoxx02 committed:
Updates

- Doc_strings
- Initial Readme

maddoxx02 committed:
Update README.md

maddoxx02 committed:
Updated Doctest

maddoxx02 committed:
Merge pull request #1 from maddoxx02/test

Doctest + DocStrings

maddoxx02 committed:
Updated docs

Missing index

maddoxx02 committed:
Added EEG Modules

maddoxx02 committed:
Updated nbs_utils.py

- Updated nbs_utils.py to include BCT_PY version of NBS - explicit threshold
- Updated example notebook

maddoxx02 committed:
Removed Pre-built docs

myxa commented on README.md:
cannot send it until Readme is completу or readable for current state of the package, so it's better to a have readme actual to what already is done rather than readme with empty holes to fill

myxa commented on tests/test_eeg_utils.py:
make sure testing eeg_utils.py is done before building up a package

maddoxx02 replied:
Yup, eeg_utils.py is ready for usage

myxa commented on tfnbs/eeg_utils.py:
not really sure about enumerating channels and frequencies; might be better to send eeg montage as argument to eegdata class, so that make it scalable in case of dealing with different eeg data. 
But that just brainstorming

**Definately** jupyter notebook example needed

maddoxx02 replied:
yes, I agree. 

Yet this version of EEG utils & code is very specific to one example. hence, left untouched

myxa commented on tfnbs/eeg_utils.py:
```diff
@@ -0,0 +1,240 @@
+        cond_prefix: Condition string for 
+        band_list : List[int], optional, frequency bands
+
+    Returns: 
```
gotta finish the docstring

myxa commented on tfnbs/nbs_utils.py:
```diff
@@ -1,2 +1,80 @@
+    ---- NEED TO VERIFY IF EXPLICIT TAIL FUNCTIONALITY IS ACCURATE 
+
+    '''
+    # Need to add conditional check for shapes of Group 1/group 2 
```
docstring needed

myxa commented on tfnbs/utils.py:
```diff
@@ -29,6 +29,10 @@ def fisher_r_to_z(r: npt.NDArray[np.float64],
         The transformation is z = arctanh(r). For r = ±1, z approaches ±infinity.
         When handle_bounds=True, these are capped at ±max_z.
+
+    >>> test is needed here?
```
yep, doctest needed

myxa commented on tfnbs/pairwise_tfns.py:
these functions, double check they are tested before building a package:
_permutation_task_ind
_permutation_task_paired


these functions need doctests or usage example and testing before building a package:
compute_permute_t_stat_diff
compute_t_stat_tfnos_diffs
compute_diffs
compute_t_stat_diff

myxa commented on tfnbs/tfnos.py:
doctests for 
get_tfce_score
get_tfce_score_scipy

myxa commented on tfnbs/utils.py:
doctests for 
fisher_z_to_r
get_components
binarize

myxa reviewed (changes requested):
Great! 
Everything is cool a little bit and we are there

keep in mind closest goals:
deploying a pip package
prereqs for that
- sphinx automated documentation -> doctests for all users functions
- availability to apply tfnos to eeg data (check this part with Irina Sergeevna)

maddoxx02 committed:
Update

- Revised DocStrings
- Added comments & descriptions where neccessary
- Improved readability

Pending:
- Add comments to TFNOS Notebook
- Add content to EEG Notebook
- Build Docs
- Send for review

maddoxx02 committed:
Updates

maddoxx02 commented:
Pending:
- add actions
- update readme with live details (i.e. email, doi etc) 
- readme image 
- usage section in documentation

maddoxx02 committed:
Update README.md

maddoxx02 committed:
Update README.md

maddoxx02 committed:
Update README.md

maddoxx02 committed:
Update Contents

- Pending Pre-defined clusters
- Test condition for NBS_Utils for Non-matching shapes of G1 & G2

maddoxx02 committed:
Update README.md
